test/src/test/java/hudson/model/ProjectTest.java

```diff
@@ -568,7 +571,12 @@ public void testDoDoDelete() throws Exception{
         auth.add(Jenkins.READ, user.getId());
         auth.add(Job.READ, user.getId());
         auth.add(Job.DELETE, user.getId());
-        List<HtmlForm> forms = j.createWebClient().login(user.getId(), "password").goTo(project.getUrl() + "delete").getForms();
+
+        JenkinsRule.WebClient wc = j.createWebClient();
+        wc.withBasicCredentials(user.getId(), "password");
```

Comment:
BTW the usage of `HudsonPrivateSecurityRealm` here is probably unnecessary; `DummySecurityRealm` would be simpler.